src/utils/cacheHandlers.ts

```diff
@@ -102,36 +63,16 @@ export const saveCacheResponse = async (response: AxiosResponse): Promise<any> =
     ...response.headers,
     [EXPIRY_HEADER_KEY]: expiresMs,
   };
-
-  const request = response.config;
-  // axios has already converted the response data according to responseType, which means that for example with
-  // json, we have an object. We must convert data back to original form before saving to cache:
-  let responseData;
-  switch (request.responseType) {
-    case 'blob':
-    case 'arraybuffer':
-    case 'text':
-      // we can save usual responses as they are:
-      responseData = response.data;
-      break;
-    case 'json':
-    case undefined: // axios defaults to json https://github.com/axios/axios#request-config
-      // but json was converted by axios to an object - and we want to save a string:
-      responseData = JSON.stringify(response.data);
-      break;
-    default:
-      throw new Error('Unsupported response type: ' + request.responseType);
-  }
-  cache.put(response.config.cacheKey, new Response(responseData, response));
+  shCache.set(response.config.cacheKey, response);
   return response;
 };
 
-const cacheStillValid = (response: Response): boolean => {
-  if (!response) {
+const cacheStillValid = (headers: Record<string, any>): boolean => {
```

Comment:
I think we should add a unit test for this as well just to ensure that we do not introduce any breaking changes in the future.

Reply:
Wouldnt hurt, added a unit test for the function `cacheStillValid`